src/ERC7540Vault.sol

```diff
@@ -234,9 +255,19 @@ contract ERC7540Vault is Auth, IERC7540 {
     }
 
     /// @inheritdoc IERC7575
+
     function deposit(uint256 assets, address receiver) external returns (uint256 shares) {
-        shares = manager.deposit(address(this), assets, receiver, msg.sender);
-        emit Deposit(msg.sender, receiver, assets, shares);
+        shares = _deposit(assets, receiver, msg.sender);
+    }
+
+    function deposit(uint256 assets, address receiver, address owner) external returns (uint256 shares) {
```

Comment:
Not quite sure why we have 2 methods. Can we leave only this one? Most of the methods rely on passing the owner. Also If for some reason there is a need for both methods, probably there should be one for redeem as well? Which these changes there is only one `redeem` method that accepts `owner`.